### DISCUSSION TITLE
Move docker directory to it's own repository

### DISCUSSION THREAD

**Discussion by ranile** [supabase/supabase#17034] [Feature Requests]
# Chore

## Describe the chore

Move `docker` directory (needed for docker setup) to it's own repository

## Additional context

This is a _very_ large repository and if someone wants to try out self-hosted docker set up, they need to clone it entirely. Cloning with `--depth 1`, I managed to let it download about ~400MBs before giving up and manually saving the files for docker set-up.

This is also a pain point for server deployments where users may not want to store the entire supabase source tree. Updating supabase also means pulling all the _source_ changes since the last update, which is also undesirable, especially in a case if an automation script is polling for updates.

**Comment by Madjarx** [comment]
Yeah i agree, having something as a standalone compose file with some empty directories serving as volumes could be okay for starters. It is sometimes PITA to clone the whole project since im most of the time just doing `dc up` and not looking too much into the rest of the code. I assume its the way most of peeps do when its self hosting anyways.

But i guess its easier for them this way to keep everything in one monorepo. I could create a repo like that for you cus i've been doing it couple times now already

**Comment by Madjarx** [comment]
https://github.com/Madjarx/supabase-docker here you go until supabase guys make something! : D

**Comment by kiwicopple** [comment]
this is mostly for us to stay organized tbh - moving it to another repo means that someone in the team need to monitor another repo and manage issues/comments

For now, i'm going to make this a Discussion - I really like the idea, but we need to consider whether it's worth it compared to some other alternatives:

- move all the "heavy assets" out of this git repo
- wrap the `docker` folder into release artifacts (so you can wget instead of git clone)

**Comment by jm-bh** [comment]
You can do a `sparse-checkout` from `git`, which has worked well for me so far.

Pulling from [this StackOverflow answer](https://stackoverflow.com/questions/600079/how-do-i-clone-a-subdirectory-only-of-a-git-repository):

```bash
git clone -n --depth=1 --filter=tree:0 https://github.com/supabase/supabase
cd supabase
git sparse-checkout set --no-cone docker
git checkout
```

Then, you can pull the latest updates with a `git pull`.

**Reply by encima** [reply]
We have added this to the [docs](https://supabase.com/docs/guides/self-hosting/docker) now. 
<img width="950" alt="image" src="https://github.com/supabase/supabase/assets/517923/f4d3a58d-ea4e-4e27-9705-22c854374bae">

Massive thanks to @cyluss for their work there!
